js/util/lru_cache.js

```diff
@@ -111,6 +111,23 @@ class LRUCache<T> {
     }
 
     /**
+     * Remove a key/value combination from the cache.
+     *
+     * @param {string} key the key for the pair to delete
+     * @returns {LRUCache} this cache
+     * @private
+     */
+    remove(key: string) {
+        if (!this.has(key)) { return this; }
+
+        this.onRemove(this.data[key]);
```

Comment:
In `add()`, it looks like `this.onRemove()` gets called after the item is deleted from `this.data`, whereas here, it's getting called before; being inconsistent _could_ cause problem at some point.

Also, a question: am I missing something, or does `cache.get()` actually also remove the item?  If that's so, is `remove()` redundant?  (Although, admittedly, I normally would _not_ expect a method called `get()` to have a side effect...)